mlrun/datastore/sources.py

```diff
@@ -313,6 +315,13 @@ def __init__(
         end_time: Optional[Union[datetime, str]] = None,
         additional_filters: Optional[list[tuple]] = None,
     ):
+        if additional_filters:
+            additional_filters_dict = {"additional_filters": additional_filters}
+            if attributes:
+                attributes = dict(attributes)
+                attributes.update(additional_filters_dict)
+            else:
+                attributes = additional_filters_dict
```

Comment:
Anyway, this is much simpler
```suggestion
            attributes = attributes or {}
            attributes["additional_filters"] = additional_filters
```

Reply:
I want to create a copy of attributes, that is why I used dict(attributes). I do not want to update original attributes.

Reply:
I changed it to something simpler:

```suggestion
            attributes = deepcopy(attributes) or {}
            attributes["additional_filters"] = additional_filters
```